### PR TITLE
feat(TBD-8684): add SSL support to Impala components

### PR DESCRIPTION
**Pull request by talend-jphautin**
**Please check if the PR fulfills these requirements**

- [X] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format) ?
- [ ] Unit tests for the Java changes have been added (for bug fixes / features) ?
- [ ] TUJ for the JavaJet changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The new code does not introduce new technical issues

**Warning**
This PR *must not* be merge until non visible group are allowed (see https://github.com/Talend/tdi-studio-se/pull/3543)

**What is the current behavior?** (You can also link to an open issue here)

Until now, Impala component does not support SSL + kerberos for cloudera (support for both was introduce in CDH 5.12 and backported to 5.7)

**What is the new behavior?**

Impala components support configuration of SSL in connection.

**BREAKING CHANGE**

If this PR contains a breaking change, please describe the impact and migration
path for existing applications.
If not please remove this section.

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.designer.components.bigdata/components/tImpalaConnection/tImpalaConnection_java.xml [hunk 1]
```diff
@@ -74,72 +98,72 @@
...
+			NUM_ROW="50" REPOSITORY_VALUE="SSL_TRUST_STORE_PASSWORD" REQUIRED="true" GROUP="ENCRYPTION"
+			SHOW_IF="(isShow[USE_SSL]) AND (USE_KRB=='false' OR DISTRIB[DISTRIBUTION, IMPALA_VERSION].doSupportSSLwithKerberos[] OR DISTRIBUTION=='CUSTOM') AND (USE_SSL=='true')">
+			<DEFAULT>"********"</DEFAULT>
+		</PARAMETER>
 	</PARAMETERS>
+	<ADVANCED_PARAMETERS>
+		<PARAMETER NAME="HADOOP_ADVANCED_PROPERTIES"
+			REPOSITORY_VALUE="HADOOP_ADVANCED_PROPERTIES" FIELD="TABLE" REQUIRED="false"
+			SHOW_IF="!DISTRIB[DISTRIBUTION, IMPALA_VERSION].useCloudLauncher[]"
+			NUM_ROW="40" NB_LINES="3">
```
Comment (CarlosTalend) [inline main/plugins/org.talend.designer.components.bigdata/components/tImpalaConnection/tImpalaConnection_java.xml]: NUM_ROW="40" between 2 "50"s. Maybe reorder.

File: main/plugins/org.talend.designer.components.bigdata/components/tImpalaCreateTable/tImpalaCreateTable_java.xml [hunk 2]
```diff
@@ -25,72 +25,138 @@
...
 			REPOSITORY_VALUE="TYPE" GROUP="CONNECTION">
 			<DEFAULT>Hive</DEFAULT>
 		</PARAMETER>
 
 		<!-- CONNECTION PART BEGIN -->
 
-		<PARAMETER NAME="USE_EXISTING_CONNECTION" FIELD="CHECK"
-			NUM_ROW="4">
+		<PARAMETER NAME="USE_EXISTING_CONNECTION" FIELD="CHECK" GROUP="CONNECTION"
+				   NUM_ROW="1">
```
Comment (CarlosTalend) [inline main/plugins/org.talend.designer.components.bigdata/components/tImpalaCreateTable/tImpalaCreateTable_java.xml]: NUM_ROW="1". Maybe reorder.

File: main/plugins/org.talend.designer.components.bigdata/components/tImpalaInput/tImpalaInput_begin.javajet [hunk 3]
```diff
@@ -19,23 +19,293 @@
...
+			boolean useSsl = "true".equals(ElementParameterParser.getValue(node, "__USE_SSL__"));
+			String sslTrustStore = ElementParameterParser.getValue(node, "__SSL_TRUST_STORE__");
+			String sslStorepasswordFieldName = "__SSL_TRUST_STORE_PASSWORD__";
+
+			boolean configureFromClassPath = "true".equals(ElementParameterParser.getValue(node, "__CONFIGURATIONS_FROM_CLASSPATH__"));
+			boolean storeByHBase = "true".equals(ElementParameterParser.getValue(node, "__STORE_BY_HBASE__"));
+		    
+
+			if(impalaDriver!=null && !"".equals(impalaDriver.trim()) && (isCustom || hiveDistrib.doSupportHive2())) {
+			    impalaDriver = impalaDriver.toLowerCase();
```
Comment (CarlosTalend) [inline main/plugins/org.talend.designer.components.bigdata/components/tImpalaInput/tImpalaInput_begin.javajet]: Ditto

File: main/plugins/org.talend.designer.components.bigdata/components/tImpalaInput/tImpalaInput_java.xml [hunk 4]
```diff
@@ -23,77 +23,141 @@
...
 			REPOSITORY_VALUE="TYPE" GROUP="CONNECTION">
 			<DEFAULT>Hive</DEFAULT>
 		</PARAMETER>
 
 		<!-- CONNECTION PART BEGIN -->
 
-		<PARAMETER NAME="USE_EXISTING_CONNECTION" FIELD="CHECK"
-			NUM_ROW="4">
+		<PARAMETER NAME="USE_EXISTING_CONNECTION" FIELD="CHECK" GROUP="CONNECTION"
+				   NUM_ROW="1">
```
Comment (CarlosTalend) [inline main/plugins/org.talend.designer.components.bigdata/components/tImpalaInput/tImpalaInput_java.xml]: Ditto

File: main/plugins/org.talend.designer.components.bigdata/components/tImpalaLoad/tImpalaLoad_java.xml [hunk 5]
```diff
@@ -19,74 +20,140 @@
...
 			REPOSITORY_VALUE="TYPE" GROUP="CONNECTION">
 			<DEFAULT>Hive</DEFAULT>
 		</PARAMETER>
 
 		<!-- CONNECTION PART BEGIN -->
 
-		<PARAMETER NAME="USE_EXISTING_CONNECTION" FIELD="CHECK"
-			NUM_ROW="4">
+		<PARAMETER NAME="USE_EXISTING_CONNECTION" FIELD="CHECK" GROUP="CONNECTION"
+				   NUM_ROW="1">
```
Comment (CarlosTalend) [inline main/plugins/org.talend.designer.components.bigdata/components/tImpalaLoad/tImpalaLoad_java.xml]: Ditto

File: main/plugins/org.talend.designer.components.bigdata/components/tImpalaOutput/tImpalaOutput_java.xml [hunk 6]
```diff
@@ -19,74 +19,141 @@
...
 			REPOSITORY_VALUE="TYPE" GROUP="CONNECTION">
 			<DEFAULT>Hive</DEFAULT>
 		</PARAMETER>
 
 		<!-- CONNECTION PART BEGIN -->
 
-		<PARAMETER NAME="USE_EXISTING_CONNECTION" FIELD="CHECK"
-			NUM_ROW="4">
+		<PARAMETER NAME="USE_EXISTING_CONNECTION" FIELD="CHECK" GROUP="CONNECTION"
+				   NUM_ROW="1">
```
Comment (CarlosTalend) [inline main/plugins/org.talend.designer.components.bigdata/components/tImpalaOutput/tImpalaOutput_java.xml]: Ditto

File: main/plugins/org.talend.designer.components.bigdata/components/tImpalaRow/tImpalaRow_java.xml [hunk 7]
```diff
@@ -22,83 +22,150 @@
...
 			REPOSITORY_VALUE="TYPE" GROUP="CONNECTION">
 			<DEFAULT>Hive</DEFAULT>
 		</PARAMETER>
 
 		<!-- CONNECTION PART BEGIN -->
 
-		<PARAMETER NAME="USE_EXISTING_CONNECTION" FIELD="CHECK"
-			NUM_ROW="4">
+		<PARAMETER NAME="USE_EXISTING_CONNECTION" FIELD="CHECK" GROUP="CONNECTION"
+				   NUM_ROW="1">
```
Comment (CarlosTalend) [inline main/plugins/org.talend.designer.components.bigdata/components/tImpalaRow/tImpalaRow_java.xml]: Ditto

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/helper/DistributionHelper.java [hunk 8]
```diff
@@ -98,10 +98,38 @@ private static Method findMethod(Class baseClazz, String methodName) {
...
+    @SuppressWarnings({ "unchecked", "rawtypes" })
+    private static Method findDefaultMethod(Class baseClazz, String methodName) {
+        Method method = null;     
+        for (Class anInterface : baseClazz.getInterfaces()) {
+            try {
+                method = anInterface.getDeclaredMethod(methodName);
+                if (method != null && method.isDefault()) {
+                    return method;
+                }
+            } catch (NoSuchMethodException | SecurityException e) {
```
Comment (CarlosTalend) [inline main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/helper/DistributionHelper.java]: Bitwise OR intended? I thought that we'd catch in any of these 2 situations

File: main/plugins/org.talend.designer.components.bigdata/components/tImpalaConnection/tImpalaConnection_begin.javajet [hunk 9]
```diff
@@ -13,40 +22,494 @@
...
+			boolean useSsl = "true".equals(ElementParameterParser.getValue(node, "__USE_SSL__"));
+			String sslTrustStore = ElementParameterParser.getValue(node, "__SSL_TRUST_STORE__");
+			String sslStorepasswordFieldName = "__SSL_TRUST_STORE_PASSWORD__";
+
+			boolean configureFromClassPath = "true".equals(ElementParameterParser.getValue(node, "__CONFIGURATIONS_FROM_CLASSPATH__"));
+			boolean storeByHBase = "true".equals(ElementParameterParser.getValue(node, "__STORE_BY_HBASE__"));
+		    
+
+			if(impalaDriver!=null && !"".equals(impalaDriver.trim()) && (this.isCustom || this.hiveDistrib.doSupportHive2())) {
+			    impalaDriver = impalaDriver.toLowerCase();
```
[the remark this answers is not in the document]
Comment (talend-jphautin) [reply main/plugins/org.talend.designer.components.bigdata/components/tImpalaConnection/tImpalaConnection_begin.javajet]: the doSupportHive2 do not check for null value so It is needed before calling it.

File: main/plugins/org.talend.designer.components.bigdata/components/tImpalaConnection/tImpalaConnection_begin.javajet [hunk 10]
```diff
@@ -13,40 +22,494 @@
...
+				}	
+				if ("impala41".equalsIgnoreCase(impalaDriver)) {
+					javaDbDriver = "com.cloudera.impala.jdbc41.Driver";
+					impalaDbProtocol="impala";
+				}
+			} else {
+				throw new IllegalArgumentException("impala can not work with Hive1");
+			}
+
+          	// When configuraing from the classpath we do not set the variables using System.setProperty
```
[the remark this answers is not in the document]
Comment (talend-jphautin) [reply main/plugins/org.talend.designer.components.bigdata/components/tImpalaConnection/tImpalaConnection_begin.javajet]: fixed

File: main/plugins/org.talend.designer.components.bigdata/components/tImpalaConnection/tImpalaConnection_java.xml [hunk 11]
```diff
@@ -48,21 +66,27 @@
...
+			SHOW="true">
 			<DEFAULT>""</DEFAULT>
 		</PARAMETER>
-
-		<PARAMETER NAME="DBNAME" FIELD="TEXT" NUM_ROW="30"
-			REPOSITORY_VALUE="SID" REQUIRED="true" GROUP="CONNECTION">
-			<DEFAULT>"default"</DEFAULT>
-		</PARAMETER>
+		
+		<PARAMETER NAME="IMPALA_ADDITIONAL_JDBC" FIELD="TEXT" NUM_ROW="29"
```
[the remark this answers is not in the document]
Comment (talend-jphautin) [reply main/plugins/org.talend.designer.components.bigdata/components/tImpalaConnection/tImpalaConnection_java.xml]: reordering it

File: main/plugins/org.talend.designer.components.bigdata/components/tImpalaConnection/tImpalaConnection_java.xml (hunk 1 above)
Comment (talend-jphautin) [reply main/plugins/org.talend.designer.components.bigdata/components/tImpalaConnection/tImpalaConnection_java.xml]: the advanced parameters are on another tab.so NUM_ROW are not related in these case IMHO.

File: main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/helper/DistributionHelper.java (hunk 8 above)
Comment (talend-jphautin) [reply main/plugins/org.talend.hadoop.distribution/src/main/java/org/talend/hadoop/distribution/helper/DistributionHelper.java]: added some comments on the logic

**Pull request merged**